### PR TITLE
regression metrics

### PR DESCRIPTION
**Pull request by lalmei**

### CODE REVIEW

File: src/whylogs/core/metrics/model_metrics.py [hunk 1]
```diff
@@ -10,22 +11,35 @@ class ModelMetrics:
 
     Attributes:
         confusion_matrix (ConfusionMatrix): ConfusionMatrix which keeps it track of counts with NumberTracker
+        regression_metrics (RegressionMetrics): Regression Metrics keeps track of a common regression metrics in case the targets are continous.
     """
 
-    def __init__(self, confusion_matrix: ConfusionMatrix = None):
-        if confusion_matrix is None:
-            confusion_matrix = ConfusionMatrix()
+    def __init__(self, confusion_matrix: ConfusionMatrix = ConfusionMatrix(),
```
Comment (andyndang) [inline src/whylogs/core/metrics/model_metrics.py]: Don't initialize object here as part of function parameters

File: src/whylogs/core/metrics/regression_metrics.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+        Returns:
+              RegressionMetrics: merged regression metrics
+        """
+
+        if self.count == 0:
+            return other_reg_met
+        if other_reg_met.count == 0:
+            return self
+
+        new_reg = RegressionMetrics()
```
Comment (andyndang) [inline src/whylogs/core/metrics/regression_metrics.py]: What about `prediction_field` etc? You also need to validate these fields match between two objects

File: tests/unit/core/metrics/test_regression_metrics.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    regmet_sum.add(df["predictions"].to_list(), df["targets"].to_list())
+
+    regmet_2 = RegressionMetrics()
+    df_2 = pd.read_parquet(os.path.join(os.path.join(TEST_DATA_PATH, "metrics", "2021-02-13.parquet")))
+    regmet_2.add(df_2["predictions"].to_list(), df_2["targets"].to_list())
+    regmet_sum.add(df_2["predictions"].to_list(), df_2["targets"].to_list())
+
+    merged_reg_metr = regmet.merge(regmet_2)
+
+    assert merged_reg_metr.count == regmet_sum.count
```
Comment (andyndang) [inline tests/unit/core/metrics/test_regression_metrics.py:70]: need to assert on field names

**Review by andyndang — CHANGES_REQUESTED** [review]

File: src/whylogs/core/metrics/regression_metrics.py (hunk 2 above)
Comment (lalmei) [reply src/whylogs/core/metrics/regression_metrics.py]: fixed in 8dc266ec0

File: tests/unit/core/metrics/test_regression_metrics.py (hunk 3 above)
Comment (lalmei) [reply tests/unit/core/metrics/test_regression_metrics.py:70]: fixed in 8dc266e

File: src/whylogs/core/metrics/model_metrics.py (hunk 1 above)
Comment (lalmei) [reply src/whylogs/core/metrics/model_metrics.py]: fixed in 8dc266e

File: src/whylogs/core/metrics/model_metrics.py [hunk 4]
```diff
@@ -54,6 +68,14 @@ def compute_confusion_matrix(self, predictions: List[Union[str, int, bool]],
             self.confusion_matrix = self.confusion_matrix.merge(
                 confusion_matrix)
 
+    def compute_regression_metrics(self, predictions: List[float],
```
Comment (andyndang) [inline src/whylogs/core/metrics/model_metrics.py]: `predictions` can be `List[int, float]` though, right?

File: src/whylogs/core/metrics/model_metrics.py [hunk 5]
```diff
@@ -10,22 +11,35 @@ class ModelMetrics:
...
         confusion_matrix (ConfusionMatrix): ConfusionMatrix which keeps it track of counts with NumberTracker
+        regression_metrics (RegressionMetrics): Regression Metrics keeps track of a common regression metrics in case the targets are continous.
     """
 
-    def __init__(self, confusion_matrix: ConfusionMatrix = None):
+    def __init__(self, confusion_matrix: ConfusionMatrix = None,
+                 regression_metrics: RegressionMetrics = None,
+                 model_type: ModelType = ModelType.UNKNOWN):
         if confusion_matrix is None:
             confusion_matrix = ConfusionMatrix()
```
Comment (andyndang) [inline src/whylogs/core/metrics/model_metrics.py]: can we have a case when both these objects are non-null and have values? I can't think of a case when both of these will live together.

I'd rather keep them separately and enforce that only one of them can be non-None at a time (and that will determine `model_type` if necessary)

**Review by andyndang — CHANGES_REQUESTED** [review]

File: src/whylogs/core/metrics/model_metrics.py (hunk 4 above)
Comment (lalmei) [reply src/whylogs/core/metrics/model_metrics.py]: very true

File: src/whylogs/core/metrics/model_metrics.py (hunk 5 above)
Comment (lalmei) [reply src/whylogs/core/metrics/model_metrics.py]: there will be cases when both will live together, but we can keep one or the other for now.

Comment (lalmei) [reply src/whylogs/core/metrics/model_metrics.py]: for now they are in two separate branches

File: src/whylogs/core/model_profile.py [hunk 6]
```diff
@@ -64,41 +67,50 @@ def compute_metrics(self, targets,
...
+                                                    target_field=target_field,
+                                                    prediction_field=prediction_field)
+            self.metrics.model_type = ModelType.REGRESSION
+
+        elif tgt_type in ("binary", "multiclass") or model_type == ModelType.CLASSIFICATION:
+            self.metrics.model_type = ModelType.CLASSIFICATION
+
+            # if score are not present set them to 1.
+            if scores is None:
+                scores = np.ones(len(targets))
```
Comment (cswarth) [inline src/whylogs/core/model_profile.py:83]: np.ones() returns an array, so does it work to wrap that in np.array() a couple lines later?

File: src/whylogs/core/model_profile.py [hunk 7]
```diff
@@ -48,14 +51,14 @@ def compute_metrics(self, targets,
         Parameters
```
Comment (cswarth) [inline src/whylogs/core/model_profile.py:51]: pydoc-markdown package we use for extracting doc comments is expecting Sphinx style documentation, but this method is using Numpy style.
I don't care which style we use but we ought to standardize on one, and the documentation for this method is not recognized and gets formatted badly.

File: src/whylogs/app/logger.py [hunk 8]
```diff
@@ -159,45 +158,44 @@ def _intialize_profiles(self,
...
-            elif m.group(2) == 'd':
-                interval *= 60 * 60 * 24  # one day
-                self.suffix = "%Y-%m-%d"
-            else:
-                raise TypeError("Invalid rotation interval, expected integer followed by one of 's', 'm', 'h', or 'd'")
-            # time in seconds
-            current_time = int(datetime.datetime.utcnow().timestamp())
-            self.interval = interval * self.interval_multiplier
-            self.rotate_at = self.rotate_when(current_time)
+        if with_rotation_time is None:
```
Comment (cswarth) [inline src/whylogs/app/logger.py:161]: We're still going to need to merge my PR to update the documentation and get the unit test for the change to with_rotation_time

**Review by cswarth — APPROVED** [review]
couple of nits but lgtm
Would like to see commits squashed

File: src/whylogs/core/model_profile.py (hunk 7 above)
Comment (cswarth) [reply src/whylogs/core/model_profile.py:51]: Don't worry about the doc change, I'll go through and standardize our comment style in one PR

**Review by andyndang — APPROVED** [review]
LGTM

File: src/whylogs/core/model_profile.py (hunk 6 above)
Comment (lalmei) [reply src/whylogs/core/model_profile.py:83]: scores may be just a list from the user side.

File: src/whylogs/app/logger.py (hunk 8 above)
Comment (lalmei) [reply src/whylogs/app/logger.py:161]: we merged that one already

**Pull request merged**